cargo-process.el

```diff
@@ -150,10 +150,11 @@
 (defun cargo-process--start (name command)
   "Start the Cargo process NAME with the cargo command COMMAND."
   (let* ((buffer (concat "*Cargo " name "*"))
+         (project-root (cargo-process--project-root))
          (path cargo-process--custom-path-to-bin)
          (path (and path (file-name-as-directory path)))
-         (command (cargo-process--maybe-read-command (concat path command)))
-         (project-root (cargo-process--project-root)))
+         (command (cargo-process--maybe-read-command
+                   (concat "cd " project-root " && " path command))))
```

Comment:
The way to run commands in a directory is to set `default-directory` so that when the command is run, it's run there, which would look like:

```elisp
(defun cargo-process--start (name command)
  "Start the Cargo process NAME with the cargo command COMMAND."
  (let* ((buffer (concat "*Cargo " name "*"))
         (path cargo-process--custom-path-to-bin)
         (path (and path (file-name-as-directory path)))
         (command (cargo-process--maybe-read-command (concat path command)))
         (project-root (cargo-process--project-root))
         (default-directory (if project-root
                                project-root
                              default-directory)))
    (save-some-buffers (not compilation-ask-about-save)
                       (lambda ()
                         (and project-root
                              buffer-file-name
                              (string-prefix-p project-root (file-truename buffer-file-name)))))
    (setq cargo-process-last-command (list name command))
    (compilation-start command 'cargo-process-mode (lambda(_) buffer))
    (set-process-sentinel (get-buffer-process buffer) 'cargo-process--finished-sentinel)))
```

With the above modification, if `project-root` isn't set for whatever reason, it leaves `default-directory` as is. I've tested this locally and it works as intended.